draft-ietf-masque-connect-ip.md

```diff
@@ -545,72 +635,71 @@ When the Context ID is set to zero, the Payload field contains a full IP packet
 
 Clients MAY optimistically start sending proxied IP packets before receiving the
 response to its IP proxying request, noting however that those may not be
-processed by the proxy if it responds to the request with a failure, or if the
-datagrams are received by the proxy before the request. Since receiving
+processed by the IP proxy if it responds to the request with a failure, or if the
+datagrams are received by the IP proxy before the request. Since receiving
 addresses and routes is required in order to know that a packet can be sent
-through the tunnel, such optimistic packets might be dropped by the proxy if it
+through the tunnel, such optimistic packets might be dropped by the IP proxy if it
 chooses to provide different addressing or routing information than what the
 client assumed.
 
-When a CONNECT-IP endpoint receives an HTTP Datagram containing an IP packet, it
+When an endpoint receives an HTTP Datagram containing an IP packet, it
```

Comment:
Shouldn't this be "IP proxy" here? The client might check the IP address (or not?) and might not send it on an outbound interface but directly forward to the application instead. Or alternative we might have to add a few "might" here...?

Reply:
Good point, added text about passing to an application